opm/simulators/wells/WellStateFullyImplicitBlackoil.hpp

```diff
@@ -817,6 +824,24 @@ namespace Opm
             return solvent_well_rate;
         }
 
+        /// One rate pr well connection.
+        std::vector<double>& perfRatePolymer() { return perfRatePolymer_; }
+        const std::vector<double>& perfRatePolymer() const { return perfRatePolymer_; }
+
+        /// One rate pr well
+        double polymerWellRate(const int w) const {
+            int connpos = 0;
+            for (int iw = 0; iw < w; ++iw) {
+                connpos += this->well_perf_data_[iw].size();
+            }
+            double polymer_well_rate = 0.0;
+            const int endperf = connpos + this->well_perf_data_[w].size();
+            for (int perf = connpos; perf < endperf; ++perf ) {
+                polymer_well_rate += perfRatePolymer_[perf];
+            }
+            return polymer_well_rate;
```

Comment:
```C++
const auto endperf = connpos + this->well_perf_data_[w].size();

return std::accumulate(&perfRatePolymer_[0] + connpos, &perfRatePolymer_[0] + endperf, 0.0);
```

Reply:
OK. I just copied these lines from the solvent code above. I can change this (and the similar lines in the solvent stuff). I assume there is some performance gain from using the std algorithms.

Reply:
> I just copied these lines from the solvent code above.

I see.

> I can change this (and the similar lines in the solvent stuff).

Yes please.

> I assume there is some performance gain from using the std algorithms.

Not really in this case.  `std::accumulate` is going to instantiate pretty much the same loop that you wrote, but there is substantial maintainability gains from using standard algorithms where possible because it communicates **intent** much more clearly than a raw loop.

Reply:
For these very simple loops, I do not see much of a read/intent advantage. However, when gradually moving towards making all parts parallel, algorithms are more easily replaced with parallel versions, so it is a bit more future-prepared.

Reply:
The code indeed looks cleaner after @bska suggestions. Thanks.